kubernetes/crd/dataset.json

```diff
@@ -0,0 +1,35 @@
+{
+	"apiVersion": "apiextensions.k8s.io/v1",
+	"kind": "CustomResourceDefinition",
+	"metadata": { "name": "datasets.new.artificialwisdom.cloud" },
+	"spec": {
+		"group": "new.artificialwisdom.cloud",
+		"versions": [{
+			"name": "v1",
```

Comment:
may I recommend `v1alpha1..3` and v1beta1..3` followed by `v1`?

Reply:
I normally just do v1, v2, v3, etc. It takes the same amount of work to write and apply a schema upgrade either way. We can also use namespacing to make last-minute changes to a schema; `datasets.stable.artificialwisdom.cloud/v1` could be slightly different. It's all a question of what we're willing to support.

Reply:
Monotonically increasing versioning is problematic because most consumers of most modern software have an expectation of (Semantic Versioning)[https://sember.org]. Its been tried, for example `systemd` and many really really really dislike the lack of `semver` in systemd` (or `kernel.org`) because it eliminates the implied compatility contract that `semver` required.

I think overloading versioning information with semantic state, such as [API or Feature Lifecycle](https://github.com/istio/community/blob/master/FEATURE-LIFECYCLE.md) is not ideal, although the alternatives presented lead to greater confusion.

If I understand your proposal, the suggestion is to decouple the semantic meaning of the `API Software Quality Lifecycle` from a counter that marks the version of the software. The main challenge I have is, why make it harder on us, and on our customers, even if they do not compensate us for the software?

The Python packaging "standard", for example, has extremely rich ways to state feature lifecycle, and continues to encode semantic information via versioning.

Why does Python do this? My take is that it is "simpler" (and certainly more crude) to look at one number, than multiple fields, to determine how ready something is for reading before cracking the cover on the booik.

In short, this approach adds needless cognitive overhead to an already cognitive demanding topic.